pkg/cloud/services/networking/port.go

```diff
@@ -303,39 +305,3 @@ func (s *Service) GarbageCollectErrorInstancesPort(eventObject runtime.Object, i
 
 	return nil
 }
-
-// collectPortSecurityGroups collects distinct securityGroups from port.SecurityGroups and port.SecurityGroupFilter fields.
-func (s *Service) collectPortSecurityGroups(portSecurityGroups []string, portSecurityGroupFilters []infrav1.SecurityGroupParam) ([]string, error) {
-	var allSecurityGroupIDs []string
-	// security groups provided with the portSecurityGroupFilters fields
-	securityGroupFiltersByID, err := s.GetSecurityGroups(portSecurityGroupFilters)
-	if err != nil {
-		return nil, fmt.Errorf("error getting security groups: %v", err)
-	}
-	allSecurityGroupIDs = append(allSecurityGroupIDs, securityGroupFiltersByID...)
-	securityGroupCount := 0
-	// security groups provided with the portSecurityGroups fields
-	allSecurityGroupIDs = append(allSecurityGroupIDs, portSecurityGroups...)
-	// generate unique values
-	uids := make(map[string]int)
-	for _, sg := range allSecurityGroupIDs {
-		if sg == "" {
-			continue
-		}
-		// count distinct values
-		_, ok := uids[sg]
-		if !ok {
-			securityGroupCount++
-		}
-		uids[sg] = 1
-	}
-	distinctSecurityGroupIDs := make([]string, 0, securityGroupCount)
-	// collect distict values
-	for key := range uids {
-		if key == "" {
-			continue
-		}
-		distinctSecurityGroupIDs = append(distinctSecurityGroupIDs, key)
-	}
-	return distinctSecurityGroupIDs, nil
```

Comment:
Do we need to retain this code which removes duplicates?

Reply:
Good question. From my understanding it was added to ensure that `securityGroupFilters` and `securityGroups` when merged didn't produce duplicates. Since we now remove the `securityGroups` it should no longer be needed. The `securityGroupFilters` are already de-duplicated [here](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/e35f1ca854b8bd7c3a06df8ff406aeb5f6c79d82/pkg/cloud/services/networking/securitygroups.go#L196).